src/cache.js

```diff
@@ -51,21 +96,65 @@ export const createCachingMethods = ({ collection, model, cache }) => {
         return EJSON.parse(cacheDoc)
       }
 
-      const doc = await loader.load(idToString(id))
+      const docs = await loader.load(JSON.stringify({ id: id }))
       if (Number.isInteger(ttl)) {
         // https://github.com/apollographql/apollo-server/tree/master/packages/apollo-server-caching#apollo-server-caching
-        cache.set(key, EJSON.stringify(doc), { ttl })
+        cache.set(key, EJSON.stringify(docs[0]), { ttl })
       }
 
-      return doc
+      return docs[0]
     },
     findManyByIds: (ids, { ttl } = {}) => {
       return Promise.all(ids.map(id => methods.findOneById(id, { ttl })))
     },
+    findByFields: async (fields, { ttl } = {}) => {
+      const cleanedFields = {}
+
+      Object.keys(fields).forEach(key => {
```

Comment:
```suggestion
      Object.keys(fields).sort().forEach(key => {
```

Because two objects can have the same keys, but `Object.keys()` might return them in a different order.